vulcanize.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+# Script to vulcanize all the Polymer files into one big blob.
+
+# Get the directory where this script is and set ROOT_DIR to that path. This
+# allows script to be run from different directories but always act on the
+# directory it is within.
+ROOT_DIR="$(cd "$(dirname $0)"; pwd)";
+STATIC_DIR="${ROOT_DIR}/ufo/static/"
+
+BOWER_OPTIONS="--allow-root --config.interactive=false"
+
+TEMP_FILE_LIST="temp_file_for_vulcanize_list"
+HTML_FILE_TO_VULCANIZE="vulcanize_input.html"
+VULCANIZED_HTML_FILE="vulcanized.html"
+
+IMPORT_BEFORE_STATEMENT='<link rel="import" href="'
+IMPORT_AFTER_STATEMENT='" />'
+
+function runAndAssertCmd ()
+{
+    echo "Running: $1"
+    echo
+    # We use set -e to make sure this will fail if the command returns an error
+    # code.
+    set -e && cd $ROOT_DIR && eval $1
+}
+
+function runInStaticDirAndAssertCmd ()
+{
+  echo "Running: $1"
+  # We use set -e to make sure this will fail if the command returns an error
+  # code.
+  if [ -d  "$STATIC_DIR" ]; then
+    echo "In: $STATIC_DIR"
+    echo
+    set -e && cd $STATIC_DIR && eval $1
+  else
+    echo "In: $ROOT_DIR"
+    echo
+    set -e && cd $ROOT_DIR && eval $1
+  fi
+}
+
+function installVulcanize ()
+{
+  runAndAssertCmd "sudo npm install -g vulcanize"
+}
+
+function updateBowerPackages ()
+{
+  runAndAssertCmd "bower install $BOWER_OPTIONS"
+}
+
+function findHtmlFilesToVulcanize ()
+{
+  runInStaticDirAndAssertCmd "rm -fr $TEMP_FILE_LIST"
+  runInStaticDirAndAssertCmd "rm -fr $HTML_FILE_TO_VULCANIZE"
+  runInStaticDirAndAssertCmd "touch $TEMP_FILE_LIST"
+  runInStaticDirAndAssertCmd "find . -name '*html' -not -name 'index.html' -not -name 'basic.html' -not -path '*test*' -not -path '*demo*' | sed 's|./||' > $TEMP_FILE_LIST"
+}
+
+function createSingleHtmlFileToVulcanize ()
+{
+  runInStaticDirAndAssertCmd "touch $HTML_FILE_TO_VULCANIZE"
+  runInStaticDirAndAssertCmd 'while read line; do echo "${IMPORT_BEFORE_STATEMENT}${line}${IMPORT_AFTER_STATEMENT}"; done < $TEMP_FILE_LIST > $HTML_FILE_TO_VULCANIZE'
```

Comment:
Okay, that helps.  But I'm scratching my head of how each `/bower_components/component_name_here/some_component.html` are being found, especially since they are deleted from each html now?